webapp/src/main/webapp/src/app/sender/archive/sender-archive.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, OnInit } from '@angular/core';
+import { select, Store } from '@ngrx/store';
+import { Subscription } from 'rxjs';
+import { BaseListingComponent } from 'src/app/shared/base-listing.component';
+import { SearchEvent } from './search-event';
+import { SenderSyncArchive } from './sender-archive';
+import { SenderArchiveService } from './sender-archive.service';
+import { SenderArchivedLoaded} from './state/sender-archive.actions';
+import { GET_SYNC_ARCHIVE } from './state/sender-archive.reducer';
+
+@Component({
+	selector: 'app-sender-archive',
+	templateUrl: './sender-archive.component.html',
+})
+export class SenderArchiveComponent extends BaseListingComponent implements OnInit {
+
+	count?: number;
+
+	loadedSubscription?: Subscription;
+
+	senderArchiveItens?: SenderSyncArchive[];
+
+	searchEvent: SearchEvent = new SearchEvent;
+
+	constructor(private service: SenderArchiveService,
+		private store: Store) {
+		super();
+	}
+
+	ngOnInit() {
+		this.init();
+
+		this.loadedSubscription = this.store.pipe(select(GET_SYNC_ARCHIVE)).subscribe(
+			countAndItems => {
+				this.count = countAndItems?.count;
+				this.senderArchiveItens = countAndItems?.items
+				this.reRender();
+			}
+		);
+
+		this.dtOptions = {
+			pagingType: 'full_numbers',
+			deferLoading: 12,
+			searching: true,
+			processing: true,
+		};
+
+		this.loadSenderArchiveData();
+	}
+
+	loadSenderArchiveData() {
+		this.service.getArchiveCountAndItems().subscribe(countAndItems => {
+			this.store.dispatch(new SenderArchivedLoaded(countAndItems));
+		});
+	}
+
+	searchByPeriod(event: Event) {
+		//Clear table content
+		this.store.dispatch(new SenderArchivedLoaded());
```

Comment:
You should instead set the count to 0 and items to an empty list rather than nothing, that way in the other function you don't have to check for undefined, this keeps the code cleaner

Reply:
//Clear table content
		let syncArchiveItens = new SenderSyncArchiveCountAndItems();
		syncArchiveItens.count = 0;
		syncArchiveItens.items = [];

		this.store.dispatch(new SenderArchivedLoaded(syncArchiveItens));

Reply:
With this i won't have undefined values.